pkg/identityserver/end_device_registry.go

```diff
@@ -151,6 +156,37 @@ func (is *IdentityServer) createEndDevice(ctx context.Context, req *ttnpb.Create
 	}
 	defer func() { is.setFullEndDevicePictureURL(ctx, dev) }()
 
+	// Store plaintext value to return in the response to clients.
+	var ptCACSecret string
+
+	if req.EndDevice.ClaimAuthenticationCode != nil {
+		ptCACSecret = req.EndDevice.ClaimAuthenticationCode.Value
+		if err = validateEndDeviceAuthenticationCode(*req.EndDevice.ClaimAuthenticationCode); err != nil {
+			return nil, err
+		}
+		if is.config.EndDevices.EncryptionKeyID != "" {
+			encrypted, err := is.KeyVault.Encrypt(
+				ctx,
+				[]byte(req.EndDevice.ClaimAuthenticationCode.Value),
+				is.config.EndDevices.EncryptionKeyID,
+			)
+			if err != nil {
+				return nil, err
+			}
+			// Store the encrypted value along with the ID of the key used to encrypt it.
+			req.EndDevice.ClaimAuthenticationCode.Value = fmt.Sprintf(
+				"%s%s%s",
+				is.config.EndDevices.EncryptionKeyID,
+				endDeviceAuthenticationCodeSeparator,
+				hex.EncodeToString(encrypted),
+			)
+		} else {
+			log.FromContext(ctx).Debug(
+				"No encryption key defined, store end device claim authentication code directly in plaintext",
```

Comment:
```suggestion
				"No encryption key defined, storing end device claim authentication code directly in plaintext",
```
Also do we really want to have this footgun?

Reply:
We use imperative tense in log messages